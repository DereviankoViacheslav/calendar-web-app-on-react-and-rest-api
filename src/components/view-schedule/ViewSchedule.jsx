import React from 'react';
import './ViewSchedule.scss';
import Sidebar from '../sidebar'
import Week from '../week';
import DayLabel from '../day-label';
import PropTypes from 'prop-types';

function ViewSchedule({ date, listEvents, onShowPopup }) {
  return (
    <>
      <DayLabel date={date} />
      <div className="schedule">
        <Sidebar />
        <Week listEvents={listEvents} date={date} onShowPopup={onShowPopup} />
      </div>
    </>
  );
}

ViewSchedule.propTypes = {
  onShowPopup: PropTypes.func.isRequired,
  listEvents: PropTypes.array,
  date: PropTypes.object.isRequired,
}

ViewSchedule.defaultProps = {
  listEvents: [],
}

export default ViewSchedule;