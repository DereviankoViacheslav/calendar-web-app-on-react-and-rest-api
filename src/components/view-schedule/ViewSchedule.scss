.schedule {
  display: flex;
  padding-top: 170px;
}