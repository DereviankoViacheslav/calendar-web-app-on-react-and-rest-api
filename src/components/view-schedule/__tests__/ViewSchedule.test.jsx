import React from 'react';
import { shallow } from 'enzyme';
import ViewSchedule from '../ViewSchedule'

describe('<ViewSchedule />', () => {
  it('Get snapshot', () => {
    const props = {
      onShowPopup: jest.fn(),
      listEvents: [],
      date: {},
    }
    const wrappedComponent = shallow(<ViewSchedule {...props} />);
    expect(wrappedComponent).toMatchSnapshot();
  });
  
  it('should render schedule', () => {
    const props = {
      onShowPopup: jest.fn(),
      listEvents: [],
      date: {},
    }
    const wrappedComponent = shallow(<ViewSchedule {...props} />);
    expect(wrappedComponent.find('.schedule').exists()).toBeTruthy();
  });

  it('should render DayLabel', () => {
    const props = {
      onShowPopup: jest.fn(),
      listEvents: [],
      date: {},
    }
    const wrappedComponent = shallow(<ViewSchedule {...props} />);
    expect(wrappedComponent.find('DayLabel').exists()).toBeTruthy();
  });

  it('should render Sidebar', () => {
    const props = {
      onShowPopup: jest.fn(),
      listEvents: [],
      date: {},
    }
    const wrappedComponent = shallow(<ViewSchedule {...props} />);
    expect(wrappedComponent.find('Sidebar').exists()).toBeTruthy();
  });
});