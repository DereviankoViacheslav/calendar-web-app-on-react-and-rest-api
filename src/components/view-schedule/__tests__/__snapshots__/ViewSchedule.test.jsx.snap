// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewSchedule /> Get snapshot 1`] = `
<Fragment>
  <DayLabel
    date={Object {}}
  />
  <div
    className="schedule"
  >
    <Sidebar />
    <Week
      date={Object {}}
      listEvents={Array []}
      onShowPopup={[MockFunction]}
    />
  </div>
</Fragment>
`;
